### PR TITLE
Adds MaxAge Property to CookieOptions and CookieBuilder

### PR DESCRIPTION
**Pull request by jkotalik**
#688 
@Tratcher Not sure if there is more to do here than this.

### CODE REVIEW

**Comment by Tratcher** [comment]
And https://github.com/aspnet/HttpAbstractions/blob/ab0185a0b8d0b7a80a6169fd78a45f00a28e057d/src/Microsoft.AspNetCore.Http/Internal/ResponseCookies.cs#L63

File: test/Microsoft.AspNetCore.Http.Tests/ResponseCookiesTest.cs [hunk 1]
```diff
@@ -1,9 +1,7 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System.Text;
 using Microsoft.AspNetCore.Http.Internal;
-using Microsoft.Extensions.ObjectPool;
 using Microsoft.Net.Http.Headers;
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Http.Tests/ResponseCookiesTest.cs]: No updated ResponseCookies test?

**Comment by jkotalik** [comment]
@Tratcher Should Max-Age be a header name in HeaderNames.cs?
\

**Comment by Tratcher** [comment]
No, it's not a header name, it's a header property name.

**Comment by jkotalik** [comment]
@Tratcher Do you recommend doing a full E2E to verify max-age shows up on requests?

**Comment by Tratcher** [comment]
on requests? None of the cookie settings are echoed on the request, on the the name and value. It's worth doing some manual testing though to make sure major browsers and HttpClient honor this setting.

**Comment by muratg** [comment]
*ping*

**Comment by jkotalik** [comment]
@Tratcher I verified locally and the max-age header is sent in the response.  :shipit: ?

**Comment by Tratcher** [comment]
Did you verify that various browsers and clients honor it? Or at least don't choke on it? HttpClient, IE, Chrome, Firefox...

File: test/Microsoft.AspNetCore.Http.Abstractions.Tests/CookieBuilderTests.cs [hunk 2]
```diff
@@ -39,6 +39,16 @@ public void ComputesExpiration()
         }
 
         [Fact]
+        public void ComputesMaxAge()
+        {
+            Assert.Null(new CookieBuilder().Build(new DefaultHttpContext()).Expires);
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Http.Abstractions.Tests/CookieBuilderTests.cs]: Why check Expires in a MaxAge test?

Comment (jkotalik) [inline test/Microsoft.AspNetCore.Http.Abstractions.Tests/CookieBuilderTests.cs]: Oops.

**Comment by muratg** [comment]
*ping* :)

File: test/Microsoft.AspNetCore.Http.Tests/ResponseCookiesTest.cs [hunk 3]
```diff
@@ -74,6 +72,23 @@ public void NoParamsDeleteRemovesCookieCreatedByAdd()
...
+            var cookieOptions = new CookieOptions();
+            var maxAgeTime = TimeSpan.FromHours(1);
+            cookieOptions.MaxAge = TimeSpan.FromHours(1);
+            var testcookie = "TestCookie";
+
+            cookies.Append(testcookie, testcookie, cookieOptions);
+
+            var cookieHeaderValues = headers[HeaderNames.SetCookie];
+            Assert.Equal(1, cookieHeaderValues.Count);
+            Assert.Contains("max-age=", cookieHeaderValues[0]);
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Http.Tests/ResponseCookiesTest.cs]: You're setting one hour, you should be able to check that value here.

**Pull request merged**